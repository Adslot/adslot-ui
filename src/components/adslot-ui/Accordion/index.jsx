import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';
import Card from 'alexandria/Card';
import Panel from '../Panel';

class Accordion extends React.PureComponent {
  static propTypes = {
    dts: PropTypes.string,
    onPanelClick: PropTypes.func,
    children: PropTypes.node,
    defaultActivePanelIds: PropTypes.arrayOf(PropTypes.string),
    maxExpand: PropTypes.oneOfType([PropTypes.number, PropTypes.oneOf(['max'])]),
  };

  static defaultProps = {
    maxExpand: 'max',
    defaultActivePanelIds: [],
  };

  state = {
    activePanelIds:
      this.props.maxExpand === 'max'
        ? this.props.defaultActivePanelIds
        : _.slice(this.props.defaultActivePanelIds, 0, this.props.maxExpand),
  };

  onPanelClick = panelId => {
    const { maxExpand } = this.props;

    if (_.includes(this.state.activePanelIds, panelId)) {
      // remove panelId out of the active list
      this.setState({ activePanelIds: _.without(this.state.activePanelIds, panelId) });
    } else {
      // drop panels from the beginning if max opened panels count is reached
      let newActivePanelIds = [...this.state.activePanelIds, panelId];
      if (maxExpand !== 'max' && newActivePanelIds.length > maxExpand) {
        newActivePanelIds = _.drop(newActivePanelIds, newActivePanelIds.length - maxExpand);
      }

      this.setState({ activePanelIds: newActivePanelIds });
    }

    if (this.props.onPanelClick) {
      this.props.onPanelClick(panelId);
    }
  };

  validateProps() {
    const { maxExpand } = this.props;

    // validate maxExpand value
    switch (true) {
      case _.isNumber(maxExpand) && maxExpand <= 0:
      case _.isString(maxExpand) && maxExpand !== 'max':
        throw new Error("maxExpand must be a positive number or 'max'");
      default:
        break;
    }
  }

  renderPanelFromChildren = child => {
    const { id, isCollapsed } = child.props;

    // prevent rendering if child is not an instance of Accordion.Panel
    if (child.type !== Panel) {
      return null;
    }

    // respects child.props.isCollapsed for controlled behaviour
    return React.cloneElement(child, {
      ...child.props,
      onClick: this.onPanelClick,
      isCollapsed: _.isNil(isCollapsed) ? !_.includes(this.state.activePanelIds, id) : isCollapsed,
    });
  };

  render() {
    const { children, dts } = this.props;
    this.validateProps();

    return (
      <Card.Container dts={dts}>
        <Card.Content fill>{React.Children.map(children, this.renderPanelFromChildren)}</Card.Content>
      </Card.Container>
    );
  }
}

Accordion.Panel = Panel;

export default Accordion;
