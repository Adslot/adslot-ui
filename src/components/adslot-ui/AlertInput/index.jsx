import _ from 'lodash';
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Popover } from 'third-party';
import './styles.scss';

export const baseClass = 'alert-input-component';

export default class AlertInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isFocused: false,
      isPopoverVisible: false,
    };
  }

  handleMouseEnter = () => {
    if (this.props.alertMessage) {
      this.setState({ isPopoverVisible: true });
    }
  };

  handleMouseLeave = () => {
    this.setState({ isPopoverVisible: false });
  };

  handleInputFocus = event => {
    event.target.select();
    this.setState({
      isFocused: true,
      isPopoverVisible: Boolean(this.props.alertMessage),
    });

    if (this.props.onFocus) {
      this.props.onFocus(event);
    }
  };

  handleInputBlur = event => {
    this.setState({
      isFocused: false,
      isPopoverVisible: false,
    });

    if (this.props.onBlur) {
      this.props.onBlur(event);
    }
  };

  render() {
    const {
      className: customClass,
      dts,
      defaultValue,
      value,
      type,
      min,
      placeholder,
      prefixAddon,
      suffixAddon,
      alertStatus,
      alertMessage,
      onValueChange,
    } = this.props;

    const className = classnames(`${baseClass}-wrapper`, customClass, {
      [alertStatus]: alertStatus,
      'is-focused': this.state.isFocused,
    });

    const popoverClassName = classnames(`${baseClass}-popover`, {
      [alertStatus]: alertStatus,
    });

    const theme = alertStatus === 'warning' ? 'warn' : alertStatus;

    return (
      <div className={baseClass}>
        <Popover
          isOpen={this.state.isPopoverVisible && !_.isEmpty(alertMessage)}
          triggers={['disabled']}
          popoverContent={<strong>{alertMessage}</strong>}
          placement="bottom"
          popoverClassNames={popoverClassName}
          theme={theme}
        >
          <div
            className={className}
            data-test-selector={dts}
            onMouseEnter={this.handleMouseEnter}
            onMouseLeave={this.handleMouseLeave}
          >
            {prefixAddon ? <span className={`${baseClass}-wrapper-addon`}>{prefixAddon}</span> : null}
            <span className={`${baseClass}-wrapper-flex-wrapper`}>
              <input
                className={`${baseClass}-wrapper-input`}
                type={type}
                defaultValue={defaultValue}
                value={value}
                min={min}
                placeholder={placeholder}
                onChange={onValueChange}
                onFocus={this.handleInputFocus}
                onBlur={this.handleInputBlur}
              />
            </span>
            {suffixAddon ? <span className={`${baseClass}-wrapper-addon`}>{suffixAddon}</span> : null}
          </div>
        </Popover>
      </div>
    );
  }
}

AlertInput.propTypes = {
  className: PropTypes.string,
  dts: PropTypes.string,
  defaultValue: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  type: PropTypes.oneOf(['text', 'number']),
  min: PropTypes.number,
  placeholder: PropTypes.string,
  prefixAddon: PropTypes.node,
  suffixAddon: PropTypes.node,
  alertStatus: PropTypes.oneOf(['success', 'info', 'warning', 'error']),
  alertMessage: PropTypes.string,
  onValueChange: PropTypes.func,
  onBlur: PropTypes.func,
  onFocus: PropTypes.func,
};

AlertInput.defaultProps = {
  type: 'text',
};
