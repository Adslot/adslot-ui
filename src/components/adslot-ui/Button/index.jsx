import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { Spinner } from 'alexandria';
import './styles.scss';

const classPrefix = 'aui-component';

class Button extends React.PureComponent {
  render() {
    const { disabled, small, isLoading, inverse, children, style, onClick } = this.props;
    const isStyledBtn = style !== 'default';
    const className = classnames([
      `${classPrefix}--btn`,
      {
        [`${classPrefix}--btn-sm`]: small,
        [`${classPrefix}--is-disabled`]: disabled,
        [`btn-is-${style}`]: isStyledBtn && !inverse,
        [`btn-is-inverse-${style}`]: isStyledBtn && inverse,
      },
    ]);

    const buttonProps = {
      className,
      disabled,
      'data-is-loading': isLoading,
      ...(disabled || isLoading ? {} : { onClick }),
    };
    return (
      <button {...buttonProps}>
        <div className="btn-label">{isLoading ? <Spinner size="small" /> : children}</div>
      </button>
    );
  }
}

Button.defaultProps = {
  small: false,
  isLoading: false,
  style: 'default',
};

Button.propTypes = {
  disabled: PropTypes.bool,
  small: PropTypes.bool,
  isLoading: PropTypes.bool,
  inverse: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
  style: PropTypes.oneOf(['primary', 'default', 'warning', 'alert', 'success']),
  onClick: PropTypes.func,
};

export default Button;
