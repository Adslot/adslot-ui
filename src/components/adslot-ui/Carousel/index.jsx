import _ from 'lodash';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Carousel from 'nuka-carousel';

require('./styles.scss');

const baseClass = 'carousel-component';
const navigationDelay = 600;
const decoratorStyles = {
  bottom: 0,
  width: '30px',
};

export const getPrevDecorator = () => {
  let previousSlideThrottled;
  const component = ({ previousSlide }) => {
    if (!previousSlideThrottled) {
      previousSlideThrottled = _.throttle(previousSlide, navigationDelay);
    }
    return (<button className={`${baseClass}-prev`} onClick={previousSlideThrottled} />);
  };
  component.propTypes = { previousSlide: PropTypes.func.isRequired };

  return {
    component,
    position: 'TopLeft',
    style: decoratorStyles,
  };
};

export const getNextDecorator = () => {
  let nextSlideThrottled;
  const component = ({ nextSlide }) => {
    if (!nextSlideThrottled) {
      nextSlideThrottled = _.throttle(nextSlide, navigationDelay);
    }
    return (<button className={`${baseClass}-next`} onClick={nextSlideThrottled} />);
  };
  component.propTypes = { nextSlide: PropTypes.func.isRequired };

  return {
    component,
    position: 'TopRight',
    style: decoratorStyles,
  };
};

class CarouselComponent extends PureComponent {
  render() {
    const { className, children } = this.props;
    const decorators = [getPrevDecorator(), getNextDecorator()];

    return (
      <Carousel decorators={decorators} {...this.props} className={classNames(baseClass, className)}>
        {children}
      </Carousel>
    );
  }
}

CarouselComponent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

// See Nuka Carousel docs for other options:
// https://github.com/FormidableLabs/nuka-carousel
CarouselComponent.defaultProps = {
  autoplay: true,
  autoplayInterval: 10000,
  cellAlign: 'center',
  cellSpacing: 15,
  slidesToShow: 2,
  slideWidth: '447.5px', // 440px (image width) + 15px (spacing) / 2
  wrapAround: true,
};

export default CarouselComponent;
