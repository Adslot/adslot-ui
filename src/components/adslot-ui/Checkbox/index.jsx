import React from 'react';
import PropTypes from 'prop-types';
import { expandDts } from 'lib/utils';

import './styles.scss';

class Checkbox extends React.Component {
  static getDerivedStateFromProps(newProps, prevState) {
    return newProps.checked === prevState.checked
      ? null
      : {
          checked: newProps.checked,
        };
  }

  constructor(props) {
    super(props);
    this.state = { checked: props.checked };
    this.onChangeDefault = this.onChangeDefault.bind(this);
  }

  onChangeDefault(event) {
    const isChecked = Boolean(event.target.checked);
    this.setState(() => ({ checked: isChecked }));
    if (this.props.onChange) {
      this.props.onChange(event, this.props.name);
    }
  }

  render() {
    const { name, value, label, dts } = this.props;
    const optional = {
      id: this.props.id ? this.props.id : null,
      className: this.props.className ? this.props.className : null,
    };
    if (this.props['data-name']) {
      optional['data-name'] = this.props['data-name'];
    }
    return (
      <div className="checkbox-component">
        <label>
          <input
            type="checkbox"
            name={name}
            value={value}
            onChange={this.onChangeDefault}
            disabled={this.props.disabled}
            checked={this.state.checked}
            {...expandDts(dts)}
            {...optional}
          />
          {label ? <span>{label}</span> : null}
        </label>
      </div>
    );
  }
}

Checkbox.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  'data-name': PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.node,
  value: PropTypes.string,
  dts: PropTypes.string,
  disabled: PropTypes.bool,
  checked: PropTypes.bool,
  onChange: PropTypes.func,
};

Checkbox.defaultProps = {
  dts: '',
  disabled: false,
  checked: false,
};

export default Checkbox;
