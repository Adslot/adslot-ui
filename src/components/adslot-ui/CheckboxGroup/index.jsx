import _ from 'lodash';
import React from 'react';
import PropTypes from 'prop-types';

const mapPropsToState = (props, state) => {
  const tempState = state;
  if (props.value) {
    _.forOwn(state.value, (value, key) => {
      tempState.value[key] = props.value.indexOf(key) !== -1;
    });
  }
  return tempState;
};

class CheckboxGroup extends React.Component {
  static getDerivedStateFromProps(nextProps, prevState) {
    return mapPropsToState(nextProps, prevState);
  }

  constructor(props) {
    super(props);
    this.state = { value: {} };
    React.Children.map(this.props.children, child => {
      this.state.value[child.props.value] = false;
    });
    if (this.props.value) {
      this.state.value = mapPropsToState(this.props, this.state.value);
    }
    this.renderChildren = this.renderChildren.bind(this);
    this.onChangeDefault = this.onChangeDefault.bind(this);
  }

  onChangeDefault(event) {
    const checkboxValue = event.target.value;
    const tempValue = _.cloneDeep(this.state.value);
    this.setState(() => {
      tempValue[checkboxValue] = !this.state.value[checkboxValue];
      if (this.props.onChange) {
        this.props.onChange(tempValue, event, this.props.name);
      }
      return { value: tempValue };
    });
  }

  renderChildren() {
    return React.Children.map(this.props.children, child =>
      React.cloneElement(child, {
        name: this.props.name,
        checked: this.state.value[child.props.value],
        onChange: (...args) => {
          if (child.props.onChange) child.props.onChange(...args);
          this.onChangeDefault(...args);
        },
      })
    );
  }

  render() {
    const classes = `checkbox-group ${this.props.className}`;
    return <div className={classes}>{this.renderChildren()}</div>;
  }
}

CheckboxGroup.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.arrayOf(PropTypes.string),
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  className: PropTypes.string,
  onChange: PropTypes.func,
};

export default CheckboxGroup;
