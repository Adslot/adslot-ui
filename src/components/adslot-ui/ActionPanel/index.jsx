import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Button from 'react-bootstrap/lib/Button';
import './styles.scss';

const ActionPanel = ({ title, className, size, onClose, children, actionButton, isModal, closeIcon }) => (
  <div className={classNames('aui--action-panel-wrapper', { 'aui--action-panel-wrapper-backdrop': isModal })}>
    <div className={classNames('aui--action-panel', className, `is-${size}`, { 'action-modal': isModal })}>
      <div className={classNames('aui--action-panel-header', { 'has-actions': actionButton })}>
        <span className="title">{title}</span>
        <span className="actions">
          <Button
            onClick={onClose}
            className={classNames('close-button', { 'close-svg-icon': !actionButton })}
            dts="header-close-button"
          >
            {actionButton ? 'Cancel' : closeIcon}
          </Button>
          {actionButton}
        </span>
      </div>
      <div className="aui--action-panel-body">{children}</div>
    </div>
  </div>
);

ActionPanel.propTypes = {
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  // large is intended to be used in a modal
  size: PropTypes.oneOf(['small', 'medium', 'large']),
  onClose: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  actionButton: PropTypes.node,
  closeIcon: PropTypes.node,
  isModal: PropTypes.bool,
};

ActionPanel.defaultProps = {
  size: 'large',
  actionButton: null,
  isModal: false,
  closeIcon: <div className="close-icon" />,
};

export default ActionPanel;
