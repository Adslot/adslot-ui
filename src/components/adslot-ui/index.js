import Accordion from 'adslot-ui/Accordion';
import AlertInput from 'adslot-ui/AlertInput';
import Carousel from 'adslot-ui/Carousel';
import ConfirmModal from 'adslot-ui/ConfirmModal';
import HelpIconPopover from 'adslot-ui/HelpIconPopover';
import FilePicker from 'adslot-ui/FilePicker';
import FormGroup from 'adslot-ui/FormGroup';
import ListPicker from 'adslot-ui/ListPicker';
import ListPickerPure from 'adslot-ui/ListPickerPure';
import PagedGrid from 'adslot-ui/PagedGrid';
import Panel from 'adslot-ui/Panel';
import Search from 'adslot-ui/Search';
import SearchBar from 'adslot-ui/SearchBar';
import SpinnerButton from 'adslot-ui/SpinnerButton';
import SplitPane from 'adslot-ui/SplitPane';
import Textarea from 'adslot-ui/Textarea';
import TextEllipsis from 'adslot-ui/TextEllipsis';
import TreePickerGrid from 'adslot-ui/TreePicker/Grid';
import TreePickerNav from 'adslot-ui/TreePicker/Nav';
import TreePickerNode from 'adslot-ui/TreePicker/Node';
import TreePickerSimplePure from 'adslot-ui/TreePicker';
import UserListPicker from 'adslot-ui/UserListPicker';
import HoverDropdownMenu from 'adslot-ui/HoverDropdownMenu';
import fastStatelessWrapper from 'adslot-ui/fastStatelessWrapper';
import InformationBox from 'adslot-ui/InformationBox';
import Nav from 'adslot-ui/Navigation';

export {
  Accordion,
  AlertInput,
  Carousel,
  ConfirmModal,
  fastStatelessWrapper,
  FilePicker,
  FormGroup,
  HelpIconPopover,
  ListPicker,
  ListPickerPure,
  Nav,
  PagedGrid,
  Panel,
  Search,
  SearchBar,
  SpinnerButton,
  SplitPane,
  Textarea,
  TextEllipsis,
  TreePickerGrid,
  TreePickerNav,
  TreePickerNode,
  TreePickerSimplePure,
  UserListPicker,
  InformationBox,
  HoverDropdownMenu,
};
