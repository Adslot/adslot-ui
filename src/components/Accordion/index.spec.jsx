import _ from 'lodash';
import { Accordion, Card } from 'adslot-ui';
import { mount, shallow } from 'enzyme';
import React from 'react';
import sinon from 'sinon';
import PanelMocks from '../Panel/mocks';

describe('AccordionComponent', () => {
  const { panel1, panel2, panel3 } = PanelMocks;

  const makeProps = override =>
    _.merge(
      {
        dts: 'my-accordion',
        onPanelClick: sinon.stub(),
        defaultActivePanelIds: [],
        maxExpand: 'max',
      },
      override
    );

  it('should render with defaults', () => {
    const wrapper = shallow(
      <Accordion {...makeProps()}>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
      </Accordion>
    );
    const cardElement = wrapper.find(Card.Content);
    expect(cardElement).to.have.length(1);
    expect(cardElement.children()).to.have.length(1);
  });

  it('should have default props', () => {
    const wrapper = shallow(
      <Accordion {...makeProps()}>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
      </Accordion>
    );
    const instance = wrapper.instance();
    expect(instance.props.defaultActivePanelIds).to.eql([]);
    expect(instance.props.maxExpand).to.equal('max');
  });

  it('should render with props', () => {
    const wrapper = shallow(
      <Accordion {...makeProps()}>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
        <Accordion.Panel {...panel2}>{panel2.content}</Accordion.Panel>
        <Accordion.Panel {...panel3}>{panel3.content}</Accordion.Panel>
      </Accordion>
    );

    const cardElement = wrapper.find(Card.Content);
    expect(cardElement).to.have.length(1);

    const panelElements = cardElement.find(Accordion.Panel);
    expect(panelElements).to.have.length(3);

    const panelElement1 = panelElements.at(0);
    expect(panelElement1.prop('id')).to.equal('1');
    expect(panelElement1.prop('onClick')).to.be.a('function');

    const panelElement2 = panelElements.at(1);
    expect(panelElement2.prop('id')).to.equal('2');
    expect(panelElement2.prop('onClick')).to.be.a('function');

    const panelElement3 = panelElements.at(2);
    expect(panelElement3.prop('id')).to.equal('3');
    expect(panelElement3.prop('onClick')).to.be.a('function');
    expect(panelElement3.prop('dts')).to.equal('panel-3'); // should generate a dts from panel id
  });

  it('should pass onPanelClick down to panels', () => {
    const props = makeProps();
    const wrapper = mount(
      <Accordion {...props}>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
        <Accordion.Panel {...panel2}>{panel2.content}</Accordion.Panel>
        <Accordion.Panel {...panel3}>{panel3.content}</Accordion.Panel>
      </Accordion>
    );
    const panelElements = wrapper.find(Accordion.Panel);

    panelElements
      .at(0)
      .find('.panel-component-header')
      .simulate('click');
    panelElements
      .at(1)
      .find('.panel-component-header')
      .simulate('click');
    panelElements
      .at(2)
      .find('.panel-component-header')
      .simulate('click');

    expect(props.onPanelClick.callCount).to.equal(3);
    expect(props.onPanelClick.firstCall.calledWith('1')).to.equal(true);
    expect(props.onPanelClick.secondCall.calledWith('2')).to.equal(true);
    expect(props.onPanelClick.thirdCall.calledWith('3')).to.equal(true);
  });

  it('should remove active panel id from active list when clicking on an expanded panel', () => {
    const wrapper = mount(
      <Accordion {...makeProps({ defaultActivePanelIds: ['1'] })}>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
        <Accordion.Panel {...panel2}>{panel2.content}</Accordion.Panel>
        <Accordion.Panel {...panel3}>{panel3.content}</Accordion.Panel>
      </Accordion>
    );
    wrapper
      .find(Accordion.Panel)
      .at(0)
      .find('.panel-component-header')
      .simulate('click');

    expect(wrapper.state()).to.eql({ activePanelIds: [] });
  });

  it('should replace active panel id when props.maxExpand is less than current opened Panels count', () => {
    const wrapper = mount(
      <Accordion {...makeProps({ defaultActivePanelIds: ['1', '2'], maxExpand: 1 })}>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
        <Accordion.Panel {...panel2}>{panel2.content}</Accordion.Panel>
        <Accordion.Panel {...panel3}>{panel3.content}</Accordion.Panel>
      </Accordion>
    );
    wrapper
      .find(Accordion.Panel)
      .at(1)
      .find('.panel-component-header')
      .simulate('click');

    expect(wrapper.state()).to.eql({ activePanelIds: ['2'] });
  });

  it('should respect isCollapsed in Panel children', () => {
    const props = makeProps();
    delete props.onPanelClick;

    const wrapper = mount(
      <Accordion {...props}>
        <Accordion.Panel {...panel1} isCollapsed>
          {panel1.content}
        </Accordion.Panel>
        <Accordion.Panel {...panel2}>{panel2.content}</Accordion.Panel>
        <Accordion.Panel {...panel3}>{panel3.content}</Accordion.Panel>
      </Accordion>
    );
    wrapper
      .find(Accordion.Panel)
      .at(0)
      .find('.panel-component-header')
      .simulate('click');

    expect(
      wrapper
        .find(Accordion.Panel)
        .at(0)
        .prop('isCollapsed')
    ).to.equal(true);
  });

  it('should throw error if props.maxExpand has invalid value', () => {
    const wrapper = shallow(
      <Accordion {...makeProps()}>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
      </Accordion>
    );

    try {
      wrapper.setProps({ maxExpand: -1 });
      wrapper.instance().validateProps();
      throw new Error('should not reach');
    } catch (err) {
      expect(err.message).to.equal("maxExpand must be a positive number or 'max'");
    }
  });

  it('should ignore children that are not an instance of Accordion.Panel', () => {
    const wrapper = mount(
      <Accordion {...makeProps()}>
        <div className="should-not-render">test</div>
        <Accordion.Panel {...panel1}>{panel1.content}</Accordion.Panel>
      </Accordion>
    );

    expect(wrapper.find('.should-not-render')).to.have.length(0);
  });
});
