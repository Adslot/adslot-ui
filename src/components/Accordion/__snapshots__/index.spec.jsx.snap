// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should have default props 1`] = `
<div
  class="card-component"
  data-test-selector="my-accordion"
  data-testid="card-container-wrapper"
>
  <div
    class="card-component-content-container"
    data-testid="card-content-container-wrapper"
  >
    <div
      class="card-component-content fill"
      data-testid="card-content-wrapper"
    >
      <div
        class="panel-component collapsed"
        data-test-selector="panel-1"
        data-testid="panel-wrapper"
      >
        <div
          class="panel-component-header clearfix"
          data-testid="panel-header"
        >
          Panel 1
        </div>
        <div
          class="panel-component-content"
          data-testid="panel-content"
        />
      </div>
    </div>
  </div>
</div>
`;
